### PR TITLE
gio: generate a bunch of auto bindings

### PR DESCRIPTION
**Pull request by bilelmoussaoui**

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
There seems to be more simple things that can be generated, will look into it in a bit :)

File: gio/src/auto/functions.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ use crate::Cancellable;
 use crate::DBusConnection;
 use crate::File;
 use crate::IOErrorEnum;
+use crate::IOModule;
```
Comment (sdroege) [inline gio/src/auto/functions.rs:11]:
```
 error[E0432]: unresolved import `crate::IOModule`
Error:   --> gio\src\auto\functions.rs:11:5
   |
11 | use crate::IOModule;
   |     ^^^^^^^^^^^^^^^ no `IOModule` in the root
```

File: gio/Gir.toml [hunk 2]
```diff
@@ -715,6 +728,11 @@ manual_traits = ["InputStreamExtManual"]
     manual = true
     doc_trait_name = "InputStreamExtManual"
 
+[[object]]
+name = "Gio.IOModule"
+status = "generate"
+cfg_condition = "unix"
```
Comment (sdroege) [inline gio/Gir.toml:734]: Why wouldn't that exist on Windows? :) That's how you get a TLS module on Windows, for example.

Comment (sdroege) [reply gio/Gir.toml:734]: But without `GTypeModule` / `GModule` bindings this is not very useful

File: gio/src/auto/datagram_based.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        unsafe {
+            from_glib_full(ffi::g_datagram_based_create_source(
+                self.as_ref().to_glib_none().0,
+                condition.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+            ))
+        }
+    }
+
+    //fn receive_messages(&self, messages: /*Ignored*/&[&InputMessage], flags: i32, timeout: i64, cancellable: Option<&impl IsA<Cancellable>>) -> Result<i32, glib::Error> {
```
Comment (sdroege) [inline gio/src/auto/datagram_based.rs:97]: These two are the two basic functions of this interface. Without it is not really useable. That's why I didn't add it yet :)

File: gio/src/auto/dtls_client_connection.rs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+use glib::signal::SignalHandlerId;
+use glib::translate::*;
+use std::boxed::Box as Box_;
+use std::fmt;
+use std::mem::transmute;
+use std::ptr;
+
+glib::wrapper! {
+    #[doc(alias = "GDtlsClientConnection")]
+    pub struct DtlsClientConnection(Interface<ffi::GDtlsClientConnection, ffi::GDtlsClientConnectionInterface>) @requires DatagramBased;
```
Comment (sdroege) [inline gio/src/auto/dtls_client_connection.rs:20]: ... and without the interface, this one (and the server part) is also not really useful.

File: gio/src/auto/enums.rs [hunk 5]
```diff
@@ -1784,6 +1871,88 @@ impl ToValue for IOErrorEnum {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Clone, Copy)]
+#[non_exhaustive]
+#[doc(alias = "GIOModuleScopeFlags")]
+pub enum IOModuleScopeFlags {
```
Comment (sdroege) [inline gio/src/auto/enums.rs:1877]: This should be a flags type, not an enum. Misses the corresponding annotation

File: gio/src/auto/functions.rs [hunk 6]
```diff
@@ -440,13 +441,17 @@ pub fn io_error_from_errno(err_no: i32) -> IOErrorEnum {
...
+    unsafe {
+        FromGlibPtrContainer::from_glib_full(ffi::g_io_modules_load_all_in_directory(
+            dirname.as_ref().to_glib_none().0,
+        ))
+    }
+}
 
 //#[doc(alias = "g_io_modules_load_all_in_directory_with_scope")]
-//pub fn io_modules_load_all_in_directory_with_scope(dirname: impl AsRef<std::path::Path>, scope: /*Ignored*/&mut IOModuleScope) -> /*Ignored*/Vec<IOModule> {
+//pub fn io_modules_load_all_in_directory_with_scope(dirname: impl AsRef<std::path::Path>, scope: /*Ignored*/&mut IOModuleScope) -> Vec<IOModule> {
```
Comment (sdroege) [inline gio/src/auto/functions.rs:454]: The scope should be easily bindable: that's just a new/free function, plus a block function

File: gio/src/auto/native_volume_monitor.rs [hunk 7]
```diff
@@ -0,0 +1,27 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use crate::VolumeMonitor;
+use std::fmt;
+
+glib::wrapper! {
+    #[doc(alias = "GNativeVolumeMonitor")]
+    pub struct NativeVolumeMonitor(Object<ffi::GNativeVolumeMonitor, ffi::GNativeVolumeMonitorClass>) @extends VolumeMonitor;
```
Comment (sdroege) [inline gio/src/auto/native_volume_monitor.rs:10]: This probably shouldn't have been in the .gir file. AFAIU it's private and exposed only via the parent class

File: gio/src/auto/socket.rs [hunk 8]
```diff
@@ -352,15 +350,6 @@ impl<O: IsA<Socket>> SocketExt for O {
         }
     }
 
-    fn condition_check(&self, condition: glib::IOCondition) -> glib::IOCondition {
```
Comment (sdroege) [inline gio/src/auto/socket.rs:355]: Why does this go away? It is still needed for non-datagram sockets, AFAIU. The two APIs are named the same but not doing exactly the same, right?

File: gio/src/auto/unix_connection.rs [hunk 9]
```diff
@@ -0,0 +1,260 @@
...
+        cancellable: Option<&impl IsA<Cancellable>>,
+        callback: P,
+    );
+
+    fn receive_credentials_async_future(
+        &self,
+    ) -> Pin<Box_<dyn std::future::Future<Output = Result<Credentials, glib::Error>> + 'static>>;
+
+    #[doc(alias = "g_unix_connection_receive_fd")]
+    fn receive_fd(&self, cancellable: Option<&impl IsA<Cancellable>>) -> Result<i32, glib::Error>;
```
Comment (sdroege) [inline gio/src/auto/unix_connection.rs:49]: That `i32` there should use one of the fd traits from std

Comment (sdroege) [reply gio/src/auto/unix_connection.rs:49]: (probably)

File: gio/src/auto/unix_connection.rs [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+    );
+
+    fn send_credentials_async_future(
+        &self,
+    ) -> Pin<Box_<dyn std::future::Future<Output = Result<(), glib::Error>> + 'static>>;
+
+    #[doc(alias = "g_unix_connection_send_fd")]
+    fn send_fd(
+        &self,
+        fd: i32,
```
Comment (sdroege) [inline gio/src/auto/unix_connection.rs:71]: And this

File: gio/Gir.toml (hunk 2 above)
Comment (bilelmoussaoui) [reply gio/Gir.toml:734]: because the windows CI failed at linking stage

File: gio/src/auto/functions.rs (hunk 1 above)
Comment (bilelmoussaoui) [reply gio/src/auto/functions.rs:11]: heh a gir issue, if a parameter is behind a feature flag; the function & its imports are not behind that feature flag

Comment (GuillaumeGomez) [reply gio/src/auto/functions.rs:11]: Please open an issue if not already done. ;)

**Comment by bilelmoussaoui** [comment]
I am going to go ahead and close this one, if some specific bindings are needed in the future we can add them separately.

**Pull request closed without merging**